ui/auth/credentials.go

```diff
@@ -0,0 +1,105 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package auth
+
+import (
+	"github.com/asdine/storm"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// Storage for Credentials
+type Storage interface {
+	StoreOrUpdate(bucket string, data interface{}) error
+	GetOneByField(bucket string, fieldName string, key interface{}, to interface{}) error
+}
+
+const (
+	initialUsername          = "myst"
+	initialPassword          = "mystberry"
+	credentialsStorageBucket = "app-credentials"
+	credentialsStorageID     = 1
+)
+
+// Credentials verifies/sets user credentials for web UI
+type Credentials struct {
+	username, password string
+	db                 Storage
+}
+
+// storeCredentials - struct stored in DB
+type storeCredentials struct {
+	ID           int
+	Username     string
+	PasswordHash string
+}
+
+// NewCredentials instance
+func NewCredentials(username, password string, db Storage) *Credentials {
+	return &Credentials{
+		username: username,
+		password: password,
+		db:       db,
+	}
+}
+
+// Validate username and password against stored Credentials
+func (credentials *Credentials) Validate() (err error) {
+	var sc *storeCredentials
+	sc, err = credentials.loadOrInitialize()
+	if err != nil {
+		return errors.Wrap(err, "could not load credentials")
+	}
+	if credentials.username != sc.Username {
+		return errors.New("bad credentials")
+	}
+	err = bcrypt.CompareHashAndPassword([]byte(sc.PasswordHash), []byte(credentials.password))
+	return errors.Wrap(err, "bad credentials")
+}
+
+func (credentials *Credentials) loadOrInitialize() (s *storeCredentials, err error) {
+	var sc = &storeCredentials{}
+	err = credentials.db.GetOneByField(credentialsStorageBucket, "ID", credentialsStorageID, sc)
+	if err == storm.ErrNotFound {
+		err = NewCredentials(initialUsername, initialPassword, credentials.db).Set()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to set initial credentials")
+		}
+		err = credentials.db.GetOneByField(credentialsStorageBucket, "ID", credentialsStorageID, sc)
+	}
+	return sc, err
+}
+
+// Set new Credentials
+func (credentials *Credentials) Set() (err error) {
+	var passwordHash []byte
+	passwordHash, err = bcrypt.GenerateFromPassword([]byte(initialPassword), bcrypt.DefaultCost)
+	if err != nil {
+		return errors.Wrap(err, "unable to generate password hash")
+	}
+	sc := &storeCredentials{
+		ID:           credentialsStorageID,
+		Username:     credentials.username,
+		PasswordHash: string(passwordHash),
+	}
+	err = credentials.db.StoreOrUpdate(credentialsStorageBucket, sc)
+	if err != nil {
```

Comment:
`if` block is not needed, it's enough to have only:
```go
return errors.Wrap(err, "unable to set credentials")
```

Reply:
It's not necessary, but it helps readability

Reply:
You have a different approach in the same file: https://github.com/mysteriumnetwork/node/pull/1144/files#diff-aedec7358933457d9e43934d625860a4R72

Reply:
I fixed that :)
I think it's ok if it's a single return, but when there are more ifs, explicit is nice. That was probably a leftover.